### DISCUSSION TITLE
Diagnostics not refreshing automatically, only after :w

### DISCUSSION THREAD

**Discussion by foxt451** [neoclide/coc.nvim#4778] [Q&A] [answered]
When I write some wrong code in a .ts file that should get highlighted and leave insert mode, nothing happens (diagnostics remain the same). They are fully updated though after :w , or after hiding the buffer and revisiting it. I looked into diagnostics list by tsserver, and I see all the diagnostics that should be displayed, but they are not highlighted. Is it a bug?

An interesting thing is that when I erase some text, or paste it without insert mode, diagnostics get updated. But when I insert it, i can't even see the old diagnostic message upon putting the cursor on the text

Here is the only config I have for coc.nvim (apart from some mappings)
```
let g:coc_global_extensions = ['coc-json', 'coc-tsserver']
let g:coc_user_config = {}
let g:coc_user_config['coc.preferences.formatOnSaveFiletypes'] = ['*']
```
<img width="657" alt="image" src="https://github.com/neoclide/coc.nvim/assets/52706110/4734e7ab-0972-435d-87cb-1c221cfecdb1">

**Comment by foxt451** [comment]
And here is a min vimrc (if that matters) when this persists
```
set nocompatible
set encoding=utf-8
set nobackup
set nowritebackup
set updatetime=300
set signcolumn=yes
set hidden
filetype plugin on
set ignorecase smartcase

call plug#begin()
Plug 'neoclide/coc.nvim', {'branch': 'release'}
call plug#end()

set undofile
set undodir=~/.vim/undo
nnoremap <F5> :MundoToggle<CR>

let g:coc_global_extensions = ['coc-json', 'coc-tsserver']

```

**Comment by fannheyward** [comment]
Reproduced with vim, nvim works as expected.

~Can't reproduce, input some _error_ code, leaving insert mode, coc-tsserver will report diagnostics, without to save or hide actions.~

<img width="463" alt="image" src="https://github.com/neoclide/coc.nvim/assets/345274/a01d7bad-2505-4d13-aa16-f3943f86d533">

**Reply by fannheyward** [reply]
After some git besect, e735a1cdfcb4e72947f1ccb73990e0bc400873ff related.

**Reply by fannheyward** [reply, marked as answer]
~https://github.com/neoclide/coc.nvim/pull/4779 can fix this~

This fix bring another issue `Cannot read properties of null (reading 'itemsMap')`.

@foxt451 I can't reproduce your issue with Vim 9.0.2100 and coc.nvim master or latest release, can you test again?
